terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/runners.ts

```diff
@@ -70,27 +70,41 @@ export function resetRunnersCaches() {
   ssmParametersCache.reset();
 }
 
+// Helper function to get the current AWS account ID
+async function getCurrentAccountId(): Promise<string> {
+  const sts = new STS();
+  const identity = await sts.getCallerIdentity().promise();
+  if (!identity.Account) {
+    throw new Error('Unable to retrieve AWS account ID');
+  }
+  return identity.Account;
+}
+
 export async function findAmiID(metrics: Metrics, region: string, filter: string, owners = 'amazon'): Promise<string> {
   const ec2 = new EC2({ region: region });
+  const accountId = await getCurrentAccountId();
+  const allOwners = [accountId, owners];
```

Comment:
Also, note that the function definition has "owners" passed in here as an optional parameter that no caller actually uses.

Consider changing that optional parameter's type to a list of strings and moving this owner identification code to a higher state of being 😇